File: frontend/kotlin/working_notes/NoteApplication/app/src/main/java/com/example/noteapplication/features/login/domain/repositories/LoginRepository.kt
package com.example.noteapplication.features.login.domain.repositories

import com.example.noteapplication.features.login.domain.models.LoginRequest
import com.example.noteapplication.features.login.domain.models.LoginResponse
import retrofit2.Call

interface LoginRepository {
    fun login(loginRequest: LoginRequest): Call<LoginResponse>?
}

File: frontend/flutter/notes_application/lib/features/login/domain/repositories/login_repository.dart
import 'package:dartz/dartz.dart';
import 'package:notes_application/core/error/failures.dart';
import 'package:notes_application/core/usecases/usecase.dart';
import 'package:notes_application/features/login/domain/entities/login_response.dart';

abstract class LoginRepository {
  Future<Either<Failure,LoginResponse>> login(LoginParams loginParams);
}